cabal-install-solver/src/Distribution/Solver/Modular/Message.hs

```diff
@@ -98,40 +159,32 @@ showMessages = go 0
               -> ConflictSet
               -> FailReason
               -> Progress Message a b
-              -> Progress String a b
+              -> Progress SummarizedMessage a b
     goPReject l qpn is c fr (Step (TryP qpn' i) (Step Enter (Step (Failure _ fr') (Step Leave ms))))
       | qpn == qpn' && fr == fr' =
         -- By prepending (i : is) we reverse the order of the instances.
         goPReject l qpn (i : is) c fr ms
     goPReject l qpn is c fr ms =
-        (atLevel l $ blurbOptions Rejecting qpn (reverse is) ++ showFR c fr)
-        (go l ms)
+        Step (SummarizedMsg $ AtLevel l $ (EntryRejectMany qpn is c fr)) (go l ms)
```

Comment:
I think that the unit test failures could have been caused by the removal of the call to `reverse` here and in `goPSkip`.

Reply:
Reinstating these `reverse` opertations as well as replacing `showQPNConflictSet` with `showOptions` seems to have fixed this. I now have zero changes with respect to `master` in the file cabal-install/tests/UnitTests/Distribution/Solver/Modular/Solver.hs